examples/tutorials/online_asr_tutorial.py

```diff
@@ -0,0 +1,161 @@
+# -----------
+#
+# Performing online speech recognition is composed of the following steps
+#
+# 1. Build the inference pipeline
+#    Emformer RNN-T is composed of three components, feature extractor,
+#    decoder and token processor.
+# 2. Format the waveform into chunks of expected sizes.
+# 3. Pass data through the pipeline.
+
+######################################################################
+# 2. Preparation
+# --------------
+#
+
+######################################################################
+#
+# .. note::
+#
+#    The streaming API requires FFmpeg libraries (>=4.1).
+#
+#    If you are using Anaconda Python distribution,
+#    ``conda install -c anaconda ffmpeg`` will install
+#    the required libraries.
+#
+#    When running this tutorial in Google Colab, the following
+#    command should do.
+#
+#    .. code::
+#
+#       !add-apt-repository -y ppa:savoury1/ffmpeg4
+#       !apt-get -qq install -y ffmpeg
+
+import os
+
+import IPython
+import torch
+import torchaudio
+
+print(torch.__version__)
+print(torchaudio.__version__)
+
+from torchaudio.prototype.io import Streamer
+
+######################################################################
+# 3. Construct the pipeline
+# -------------------------
+#
+# Pre-trained model weights and related pipeline components are
+# bundled as :py:func:`torchaudio.pipelines.RNNTBundle`.
+#
+# We use :py:func:`torchaudio.pipelines.EMFORMER_RNNT_BASE_LIBRISPEECH`,
+# which is a Emformer RNN-T model trained on LibriSpeech dataset.
+#
+
+bundle = torchaudio.pipelines.EMFORMER_RNNT_BASE_LIBRISPEECH
+
+decoder = bundle.get_decoder()
+token_processor = bundle.get_token_processor()
+feature_extractor = bundle.get_streaming_feature_extractor()
+
+######################################################################
+# Streaming inference works on input data with overlap.
+# Emformer RNN-T expects right context like illustrated like the
+# following.
+#
+# .. image:: https://download.pytorch.org/torchaudio/tutorial-assets/emformer_rnnt_context.png
+#
+# The size of main segment and right context can be retrieved from bundle.
+#
+
+frames_per_chunk = bundle.segment_length * bundle.hop_length
+right_context_size = bundle.right_context_length * bundle.hop_length
+
+print(f"Main segment size: {frames_per_chunk} frames ({frames_per_chunk / bundle.sample_rate} seconds)")
+print(f"Right context size: {right_context_size} frames ({right_context_size / bundle.sample_rate} seconds)")
+
+######################################################################
+# 4. Configure the audio stream
+# -----------------------------
+#
+# Next, we configure the input audio stream using :py:func:`~torchaudio.prototype.io.Streamer`.
+#
+# For the detail of this API, please refer to the
+# `Media Stream API tutorial <../tutorials/streaming_api_tutorial.html>`__.
+#
+
+src = (
+    "https://ia800302.us.archive.org/33/items/"
+    "great_pirate_stories_0905_librivox.org/greatpiratestories_00_various.mp3"
+)
```

Comment:
Shall we add the reference to the mp3 file?

Reply:
yup. done.